src/main/java/org/jabref/gui/groups/DeleteDialog.java

```diff
@@ -0,0 +1,125 @@
+package org.jabref.gui.groups;
+
+import javax.swing.JButton;
+import javax.swing.JDialog;
+import javax.swing.JPanel;
+import javax.swing.KeyStroke;
+
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.groups.GroupTreeNode;
+import javafx.collections.ObservableList;
+
+import javax.swing.BoxLayout;
+import javax.swing.JLabel;
+import javax.swing.JComponent;
+
+import java.awt.event.ActionListener;
+import java.awt.event.KeyEvent;
+import java.util.Iterator;
+import java.util.Set;
+import java.awt.event.ActionEvent;
+
+public class DeleteDialog extends JDialog {
```

Comment:
Can you please use JavaFX to implement this dialog. We already have a quite extensive `DialogService` class that you can use.